sklearn/decomposition/sparse_pca.py

```diff
@@ -231,6 +245,57 @@ def transform(self, X, ridge_alpha='deprecated'):
 
         return U
 
+    def _get_explained_variance(self, X):
+        '''
+            Get the explained variance from the principal components of the
+            data.
+
+            Parameters
+            ----------
+            X : ndarray, shape (n_samples, n_features)
+                The feature vector. n_samples and n_features are the number of
+                samples and features, respectively.
+
+            Notes
+            -----
+            The variance ratio may not be computed. The main reason is that we
+            do not know what the total variance is since we did not compute all
+            the components.
+        '''
+        components = self.components_
+        # requires the mean to be subtracted
+        if not self.normalize_components:
+            # should be removed in version 0.22
+            X = X - np.mean(X, axis=0)
+        # the number of samples
+        n_samples = X.shape[0]
+        n_components = components.shape[0]
+        unit_vecs = components.copy()
+        components_norm = \
```

Comment:
We tend to avoid this kind of line continuation and prefer breaking after a comma within parentheses

Reply:
removed these awkward indents, thanks!